docs/src/main/sphinx/admin/properties-http-server.md

```diff
@@ -0,0 +1,155 @@
+API](/develop/client-protocol).
+
+## General
+
+### `http-server.process-forwarded`
+
+- **Type:** [](prop-type-boolean)
+- **Default value:** `false`
+
+Enable treating forwarded HTTPS requests over HTTP as secure. Requires the
+`X-Forwarded-Proto` header to be set to `HTTPS` on forwarded requests. This is
+commonly performed by a load balancer that terminates HTTPS to HTTP. Set to
+`true` when using such a load balancer in front of Trino or [Trino
+Gateway](https://trinodb.github.io/trino-gateway/). Find more details in
+[](https-load-balancer).
+
+## HTTP and HTTPS
+
+### `http-server.http.port`
+
+- **Type:** [](prop-type-integer)
+- **Default value:** `8080`
+
+Specify the HTTP port for the HTTP server.
+
+### `http-server.https.enabled`
+
+- **Type:** [](prop-type-boolean)
+- **Default value:** `false`
+
+Enable [](/security/tls).
+
+### `http-server.https.port`
+
+- **Type:** [](prop-type-integer)
+- **Default value:** `8443`
+
+Specify the HTTPS port for the HTTP server.
+
+### `http-server.https.included-cipher` and `http-server.https.excluded-cipher`
+
+Optional configuration for ciphers to use TLS, find details in
+[](tls-version-and-ciphers).
+
+### `http-server.https.keystore.path`
+
+- **Type:** [](prop-type-string)
+
+The location of the PEM or Java keystore file used to enable [](/security/tls).
+
+### `http-server.https.keystore.key`
+
+- **Type:** [](prop-type-string)
+
+The password for the PEM or Java keystore.
+
+### `http-server.https.truststore.path`
+
+- **Type:** [](prop-type-boolean)
+- **Default value:** `false`
+
+The location of the optional PEM or Java truststore file for additional
+certificate authorities. Find details in [](/security/tls).
+
+### `http-server.https.truststore.key`
+
+- **Type:** [](prop-type-boolean)
+- **Default value:** `false`
+
+The password for the optional PEM or Java truststore.
+
+### `http-server.https.keymanager.password`
+
+- **Type:** [](prop-type-string)
+
+Password for a key within a keystore, when a different password is configured
+for the specific key. Find details in [](/security/tls).
+
+### `http-server.https.secure-random-algorithm`
+
+- **Type:** [](prop-type-string)
+
+Optional name of the algorithm to generate secure random values for
+[internal communication](internal-performance).
+
+### `http-server.https.ssl-session-timeout`
+
+- **Type:** [](prop-type-duration)
+- **Default value:** `4h`
+
+Time duration for a valid TLS client session.
+
+### `http-server.https.ssl-session-cache-size`
+
+- **Type:** [](prop-type-integer)
+- **Default value:** `10000`
+
+Maximum number of SSL session cache entries.
+
+### `http-server.https.ssl-context.refresh-time`
+
+- **Type:** [](prop-type-duration)
+- **Default value:** `1m`
+
+Time between reloading default certificates.
+
+## Authentication
+
+### `http-server.authentication.type`
+
+- **Type:** [](prop-type-string)
+
+Configures the ordered list of enabled
+[authentication types](/security/authentication-types).
```

Comment:
This a comma-separated list iirc? Might be worth confirming here 👍

Reply:
Actually scratch that, the linked page is very clear already, no need to double-up 👍

Reply:
Haha .. yeah .. that was my thinking .. also hence using "authentication types" (plural)